libs/types-common/src/Data/Id.hs

```diff
@@ -112,20 +114,30 @@ newtype Id a
 -- REFACTOR: non-derived, custom show instances break pretty-show and violate the law
 -- that @show . read == id@.  can we derive Show here?
 instance Show (Id a) where
-  show = toString . toUUID
+  show = UUID.toString . toUUID
 
 instance Read (Id a) where
   readsPrec n = map (\(a, x) -> (Id a, x)) . readsPrec n
 
 instance FromByteString (Id a) where
   parser = do
-    x <- takeByteString
-    case fromASCIIBytes x of
+    x <-
+      -- we only want the matching ByteString
```

Comment:
lacking context here.  do you mean the matching *part* of the bytestring?  if so that sounds odd to me.

also, one-letter variable name?  (sorry :))

Reply:
Yes, only the matching part. This is because if we compose the parser with e.g. the domain parser to parse a qualified ID like `61a73a52-e526-4892-82a9-3d638d77629f@example.com`, the parser for the UUID should not consume that whole string, but leave the `@example.com` for the next parser.

I'll add that explanation.